spyder/plugins/completion/kite/plugin.py

```diff
@@ -167,8 +167,19 @@ def _kite_onboarding(self):
         """Request the onboarding file."""
         # No need to check installed status,
         # since the get_onboarding_file call fails fast.
-        if self._show_onboarding:
-            self.client.sig_perform_onboarding_request.emit()
+        if not self.enabled:
+            return
+        if not self._show_onboarding:
+            return
+        if self.main.is_setting_up:
+            return
+        if not self.available_languages:
+            return
+        if not self._show_onboarding:
+            return
```

Comment:
This condition is repeated, i.e. `if not self._show_onboarding` is also above.

Reply:
good catch! fixed